[CI] StatelessClusterIntegrityStressIT testRandomActivities failing

**Build Scans:**
- [elasticsearch-serverless-validate-submodule #20090 / java_run_serverless_elasticsearch_tests](https://gradle-enterprise.elastic.co/s/ejclkjsvfm4te)
- [elasticsearch-serverless-es-pr-check #58365 / java_checkpart1](https://gradle-enterprise.elastic.co/s/5zh6iy6aoof5k)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.commits.StatelessClusterIntegrityStressIT.testRandomActivities" -Dtests.seed=CCA04A1444083D8F -Dtests.locale=as-IN -Dtests.timezone=America/Costa_Rica -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(co.elastic.elasticsearch.stateless.commits.StatelessClusterIntegrityStressIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRandomActivities'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: unexpected
```

**Issue Reasons:**
- [main] 2 failures in test testRandomActivities (1.6% fail rate in 128 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

---

Migrated from https://github.com/elastic/elasticsearch-serverless/issues/5278

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -184,9 +184,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.cache.StatelessOnlinePrewarmingIT
   method: testShardPrewarming
   issue: https://github.com/elastic/elasticsearch/issues/147838
-- class: org.elasticsearch.xpack.stateless.commits.StatelessClusterIntegrityStressIT
-  method: testRandomActivities
-  issue: https://github.com/elastic/elasticsearch/issues/147841
 - class: org.elasticsearch.xpack.stateless.autoscaling.search.AutoscalingReplicaIT
   method: testSearchSizeAffectsReplicasSPBetween100And250
   issue: https://github.com/elastic/elasticsearch-serverless/issues/5731

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/commits/StatelessClusterIntegrityStressIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/commits/StatelessClusterIntegrityStressIT.java
@@ -8,6 +8,7 @@
 package org.elasticsearch.xpack.stateless.commits;
 
 import org.apache.logging.log4j.Level;
+import org.apache.lucene.store.AlreadyClosedException;
 import org.elasticsearch.action.admin.cluster.health.ClusterHealthRequest;
 import org.elasticsearch.action.admin.cluster.health.ClusterHealthResponse;
 import org.elasticsearch.action.admin.indices.flush.FlushRequestBuilder;
@@ -722,7 +723,11 @@ public class StatelessClusterIntegrityStressIT extends AbstractStatelessPluginIn
                             }
                         };
                         masterClusterService.addListener(clusterStateListener);
-                        indexShard.failShard("broken", new Exception("boom local " + (searchShard ? "search" : "indexing") + " shard"));
+                        try {
+                            indexShard.failShard("broken", new Exception("boom local " + (searchShard ? "search" : "indexing") + " shard"));
+                        } catch (AlreadyClosedException e) {
+                            logger.info("--> shard is already closed due to concurrent node restart/replace/isolation");
+                        }
                         safeAwait(unassignedLatch);
                         masterClusterService.removeListener(clusterStateListener);
                         ensureGreenIgnoreNodeDisconnection(0, trackedIndex.index.getName());

